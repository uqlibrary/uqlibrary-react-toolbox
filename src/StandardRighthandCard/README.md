# StandardRighthandCard

component for a right hand panel wrapper

## Props

- `title: PropTypes.string.isRequired`
- `help: PropTypes.object`, eg {title: 'About', text: 'help text..', buttonLabel:'Close'}

## Usage

**./App.js**
```jsx
import React from 'react';
...

import {StandardRighthandCard} from 'uqlibrary-react-toolbox';
        
const App = () => (
  <div>
    <StandardRighthandCard title='Title' help={locale.components.somthing.help}>
      Content
    </StandardRighthandCard>
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```

