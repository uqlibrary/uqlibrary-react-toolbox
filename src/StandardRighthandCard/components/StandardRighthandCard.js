import React from 'react';
import PropTypes from 'prop-types';
import {HelpIcon} from '../../index';

export default function StandardRighthandCard({title, children, help}) {
    return (
        <div className="StandardRighthandCard">
            <div className="columns is-gapless is-mobile facetsTitle">
                <div className="column">
                    {title && <h2 className="title is-5">{title}</h2>}
                </div>
                {help && help.text &&
                <div className="column is-narrow is-helpicon">
                    <HelpIcon {...help}/>
                </div>
                }
            </div>
            {children}
        </div>
    );
}

StandardRighthandCard.propTypes = {
    children: PropTypes.any,
    title: PropTypes.string,
    help: PropTypes.shape({
        title: PropTypes.string,
        text: PropTypes.any,
        buttonLabel: PropTypes.string
    })
};
