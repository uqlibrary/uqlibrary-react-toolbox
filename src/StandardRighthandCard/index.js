export StandardRighthandCard from './components/StandardRighthandCard';
