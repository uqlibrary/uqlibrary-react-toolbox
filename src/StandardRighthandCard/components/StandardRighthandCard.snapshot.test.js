jest.dontMock('./StandardRighthandCard');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import StandardRighthandCard from './StandardRighthandCard';

function setup(title, help, children) {
    const props = {title, help, children};
    return shallow(<StandardRighthandCard {...props} />);
}

describe('Snapshot tests for StandardRighthandCard component', () => {
    it('renders StandardRighthandCard with title and no help icon', () => {
        const wrapper = setup('card title');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders StandardRighthandCard with title and help button', () => {
        const wrapper = setup('Title', {title:'Help text', text:'Some help text', buttonLabel:'OK'});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders StandardRighthandCard with title and help button and some content', () => {
        const wrapper = setup('Title', {title:'Help text', text:'Some help text', buttonLabel:'OK'}, 'Some content');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});