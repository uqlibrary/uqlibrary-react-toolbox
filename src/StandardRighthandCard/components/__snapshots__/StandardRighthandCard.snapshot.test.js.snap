// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for StandardRighthandCard component renders StandardRighthandCard with title and help button 1`] = `
<div
  className="StandardRighthandCard"
>
  <div
    className="columns is-gapless is-mobile facetsTitle"
  >
    <div
      className="column"
    >
      <h2
        className="title is-5"
      >
        Title
      </h2>
    </div>
    <div
      className="column is-narrow is-helpicon"
    >
      <Connect(HelpIcon)
        buttonLabel="OK"
        text="Some help text"
        title="Help text"
      />
    </div>
  </div>
</div>
`;

exports[`Snapshot tests for StandardRighthandCard component renders StandardRighthandCard with title and help button and some content 1`] = `
<div
  className="StandardRighthandCard"
>
  <div
    className="columns is-gapless is-mobile facetsTitle"
  >
    <div
      className="column"
    >
      <h2
        className="title is-5"
      >
        Title
      </h2>
    </div>
    <div
      className="column is-narrow is-helpicon"
    >
      <Connect(HelpIcon)
        buttonLabel="OK"
        text="Some help text"
        title="Help text"
      />
    </div>
  </div>
  Some content
</div>
`;

exports[`Snapshot tests for StandardRighthandCard component renders StandardRighthandCard with title and no help icon 1`] = `
<div
  className="StandardRighthandCard"
>
  <div
    className="columns is-gapless is-mobile facetsTitle"
  >
    <div
      className="column"
    >
      <h2
        className="title is-5"
      >
        card title
      </h2>
    </div>
  </div>
</div>
`;
